ui/packages/shared/functions/src/hooks/useURLState.tsx

```diff
@@ -0,0 +1,60 @@
+// Copyright 2022 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {useState, useEffect, Dispatch} from 'react';
+import {useNavigate} from 'react-router-dom';
+import {parseParams, convertToQueryParams} from '../index';
+import {useAppDispatch, useAppSelector, setGlobalStateValue} from '@parca/store';
+
+// every time the param value changes, the URL is updated
+export const useURLState = (paramName: string, withURLUpdate: boolean = true) => {
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+  const router = parseParams(window.location.search);
+  const storeValue = useAppSelector(state => state.profile[paramName]);
```

Comment:
Shall we move this selector into the slice and import it from there so that the select logic is not dispersed across components?

Similar to this: https://github.com/parca-dev/parca/blob/9efbbdb380472db6b27702da2a7a61e341352af0/ui/packages/shared/store/src/slices/profileSlice.ts#L51